npc/woe-se/agit_main_se.txt

```diff
@@ -1700,6 +1704,33 @@ OnBarrierDestroyed:
 	end;
 
 OnDisable:
+	if (compare(strnpcinfo(NPC_NAME_HIDDEN),"arug")) {
+		if (strnpcinfo(NPC_NAME_HIDDEN) == "arug_cas01") {
+			setarray(.@x[0], 239, 241, 243, 245);
+			setarray(.@y[0],  73,  73,  73,  73);
+		} else if (strnpcinfo(NPC_NAME_HIDDEN) == "arug_cas02") {
+			setarray(.@x[0], 137, 139, 141, 143);
+			setarray(.@y[0], 137, 137, 137, 137);
+		} else {	// Castles 3, 4, 5 are identical.
+			setarray(.@x[0], 139, 141, 143, 145);
+			setarray(.@y[0], 111, 111, 111, 111);
+		}
+	} else {
+		if (strnpcinfo(NPC_NAME_HIDDEN) == "schg_cas02") {
+			setarray(.@x[0], 289, 289, 289, 289);
+			setarray(.@y[0],  98, 100, 102, 104);
+		} else if (strnpcinfo(NPC_NAME_HIDDEN) == "schg_cas03") {
+			setarray(.@x[0], 326, 328, 330);
+			setarray(.@y[0], 300, 300, 300);
+		} else {	// Castles 1, 4, 5 are identical.
+			setarray(.@x[0], 115, 117, 119, 121, 123, 125);
+			setarray(.@y[0],  49,  49,  49,  49,  49,  49);
+			setarray(.@x2[0], 115, 117, 119, 121, 123, 125);
+			setarray(.@y2[0],  50,  50,  50,  50,  50,  50);
+			setcell(strnpcinfo(NPC_NAME_HIDDEN), .@x2[0], .@y2[0], .@x2[5], .@y2[5], cell_basilica, false);
```

Comment:
You are totally right, those index aren't used. I did that following previous setarray but it isn't needed in there. I will change it, thanks :)